### PR TITLE
Commander

### PR DESCRIPTION
**Pull request by ranm8**
## Proposed Changes

-
-
-

### CODE REVIEW

File: components/ui/command-bar/autocomplete-input/autocomplete-input.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect, createRef, InputHTMLAttributes } from 'react';
+import classNames from 'classnames';
+import styles from './autocomplete-input.module.scss';
+
+type AutoCompleteInputProps = InputHTMLAttributes<HTMLInputElement> & {
+  focus?: Truthy;
+};
+
+type Truthy = boolean | number | string | null | undefined;
```
Comment (GiladShoham) [inline components/ui/command-bar/autocomplete-input/autocomplete-input.tsx:9]: It's really confusing to have type called Truthy that can get null and undefined:)

File: components/ui/command-bar/command-bar.docs.mdx [hunk 2]
```diff
@@ -0,0 +1,110 @@
+Command bar is a search box with built in results area, and keyboard navigation.
```
Comment (GiladShoham) [inline components/ui/command-bar/command-bar.docs.mdx:1]: why you don't have the metadata section?

File: components/ui/command-bar/autocomplete-input/autocomplete-input.tsx (hunk 1 above)
Comment (KutnerUri) [reply components/ui/command-bar/autocomplete-input/autocomplete-input.tsx:9]: truthy as in https://developer.mozilla.org/en-US/docs/Glossary/Truthy

**Comment by KutnerUri** [comment]
@GiladShoham - Ran is supposed to tag and export the command bar, so half of this will get removed

File: components/ui/command-bar/search-result.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ export type Keybinding = string | string[]; // TODO
 
 export type SearchResult = {
   id: string;
-  action: Function;
+  action: Function|string;
```
Comment (KutnerUri) [inline components/ui/command-bar/search-result.ts:7]: string action?

File: scopes/preview/ui/component-preview/preview.tsx [hunk 4]
```diff
@@ -55,5 +55,5 @@ export function ComponentPreview({
 
   const url = toPreviewUrl(component, previewName, queryParams);
 
-  return <iframe {...rest} ref={iframeRef} style={{ ...style, height }} src={url} />;
+  return <iframe {...rest} ref={iframeRef} style={{ ...style, height: height || 'calc(100vh + 200px)' }} src={url} />;
```
Comment (KutnerUri) [inline scopes/preview/ui/component-preview/preview.tsx:58]: this is bad - it is larger than the screen, and it is bad defaults.

--> apply it in the specific consumer

File: scopes/ui-foundation/uis/side-bar/component-tree/component-tree.tsx [hunk 5]
```diff
@@ -14,20 +14,23 @@ type ComponentTreeProps = {
...
   TreeNode = DefaultTreeNodeRenderer,
 }: ComponentTreeProps) {
   const { pathname = '/' } = useLocation() || {};
 
   const activeComponent = useMemo(() => {
     const componentUrlRegex = new RegExp(componentIdUrlRegex);
     const path = pathname?.startsWith('/') ? pathname.substring(1) : pathname;
-    const matcher = path.match(componentUrlRegex)?.[0]; // returns just the part that matches the componentId section without /~compositions etc.
+    const rawMatcher = path.match(componentUrlRegex)?.[0]; // returns just the part that matches the componentId section without /~compositions etc.
+    const matcher = assumeScopeInUrl ? rawMatcher?.split('/').slice(2).join('/'): rawMatcher;
```
Comment (KutnerUri) [inline scopes/ui-foundation/uis/side-bar/component-tree/component-tree.tsx:33]: better use `ComponentID`

**Pull request merged**